api/godog/retry/retry.go

```diff
@@ -0,0 +1,34 @@
+package retry
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+type Until struct {
+	Condition func(body []byte) bool
+	Timeout   time.Duration
+}
+
+func Try(function func() (bool, error), timeout time.Duration) error {
+	for i := 0; i < int(timeout.Milliseconds()); i++ {
+		success, err := function()
+		if err != nil {
+			return err
+		}
+
+		if success {
+			return nil
+		}
+
+		sleepDuration := time.Duration(math.Pow(2, float64(i))*100) * time.Millisecond // nolint: gomnd
+		if sleepDuration > timeout {
+			break
+		}
+
+		time.Sleep(sleepDuration)
```

Comment:
Have a look at time.After it might be possible to do something fancy with it ;)

Reply:
Good tip :) You're thinking using time.After as a timer instead of appending to timeSlept (in the current implementation)?

Reply:
Maybe there is no reason for being fancy since we don't have anything else to do in the meantime.

Reply:
mainly channel awareness :) I'm having a hard time finding use cases for them myself :)